waspc/src/Wasp/Generator/ConfigFileGenerator.hs

```diff
@@ -0,0 +1,40 @@
+module Wasp.Generator.ConfigFileGenerator
+  ( isTailwindUsed,
+    genConfigFiles,
+  )
+where
+
+import Data.List (find, isSuffixOf)
+import Data.Maybe (isJust)
+import StrongPath (File', Path', Rel)
+import qualified StrongPath as SP
+import Wasp.AppSpec (AppSpec)
+import qualified Wasp.AppSpec as AS
+import Wasp.Common (WaspProjectDir)
+import qualified Wasp.ConfigFile as CF
+import Wasp.Generator.FileDraft (FileDraft, createCopyFileDraft)
+import Wasp.Generator.Monad (Generator)
+
+-- | Helper that determines if Tailwind used. For our purposes, we allow
+-- developers to opt-in by creating both a tailwind config file and
+-- postcss config file in their wasp project dir.
+isTailwindUsed :: AppSpec -> Bool
+isTailwindUsed spec =
+  doesConfigFileExist spec CF.tailwindConfigFile
+    && doesConfigFileExist spec CF.postcssConfigFile
+
+-- | Internal helper for determining if some wasp project file is a tracked config file in AppSpec.
+doesConfigFileExist :: AppSpec -> Path' (Rel WaspProjectDir) File' -> Bool
+doesConfigFileExist spec file =
+  isJust $
+    find
+      (\f -> SP.fromRelFile file `isSuffixOf` SP.fromAbsFile (CF._pathInWaspDir f))
```

Comment:
Hmm, I will have to think on that a bit more. I suppose it could find some config file lower than where we expect it to be. I think changing it to be relative to waspDir complicates things since I don't have that waspDir info where I need it in the generator (I will add more to that comment), so I may try to think of a safer solution here to make sure it is at the root, thanks.

Reply:
While not enforced by the type system (boo!), I think in practice we are ok here. When we "discover" config files, we construct their absolute path from the wasp dir:

```haskell
-- | Discovers config files of interest in the wasp project dir.
discoverConfigFiles :: Path' Abs (Dir WaspProjectDir) -> ConfigFileRelocationMap -> IO [ConfigFileRelocator]
discoverConfigFiles waspDir configFileRelocationMap = do
  files <- fst <$> Util.IO.listDirectory waspDir
  return $ mapMaybe fileToMaybeConfigFileRelocator files
  where
    fileToMaybeConfigFileRelocator :: Path' (Rel WaspProjectDir) File' -> Maybe ConfigFileRelocator
    fileToMaybeConfigFileRelocator file = do
      projectRootDirPath <- Data.lookup file configFileRelocationMap
      return $
        ConfigFileRelocator
          { _pathInWaspDir = waspDir </> file, 👈
            _pathInProjectRootDir = projectRootDirPath
          }
```

So when we ask is this relative file from the wasp dir a suffix to an abs file that includes the wasp dir, we should be ok on the lookup so long as we did not add 2 identical files to the _mapping table_ here:

```haskell
configFileRelocationMap :: ConfigFileRelocationMap
configFileRelocationMap =
  fromList
    [ (tailwindConfigFile, asProjectRootDirConfigFile tailwindConfigFile),
      (postcssConfigFile, asProjectRootDirConfigFile postcssConfigFile)
    ]
```

At this point, the odds of us needing/doing that are low, but perhaps we should still account for this now. One way we could account for it is to add a new field to the `ConfigFileRelocator` type. Perhaps we track the relative path from wasp dir (in addition to the absolute used during generation), or maybe we allow for some string identifier to use for lookups? 🤔

Reply:
It still seems to me like there could be a mistake here -> what if we have "config.json" file in the mapping, but user has "myconfig.json" in their wasp dir -> aren't we going to detect that as a match and say that config file exists, although it doesn't?

I think the only right way is to know the relative path.
We can do that by additionally storing that to the ConfigFileRelocator, that is correct! So it would have both abs path + rel path. But then we could have just stored abs path to wasp dir + rel path to wasp dir inside it? That sounds bad because then on multiple places we have the same thing (abs path to wasp dir) and that is duplication of data, but it comes to the same thing at the end. We could maybe have tuple (abs path to wasp dir, rel path from wasp dir to config file).

Or, we add abs path to wasp dir into AppSpec. That sounds like the best solution, doesn't it? We don't have to move the whole code to use it right now, we can use it just here.

Let me know if I am missing something here and you think I am overdoing this, but to me it doesn't feel right at the moment to let it be like this.

Reply:
Hey there, yeah once we start doing arbitrary configs it does become problematic, I agree. Let's fix it now and good catch again.

I think adding abs path to wasp dir to AppSpec also sounds like the cleanest solution. For some reason, the fact that it did not include it already made me a bit worried about doing so, but I am not sure why lol. It solves this problem and probably some other past/future ones.

I can add that, make use of it here, and maybe just add a TODO above it to refactor other abs file paths to make use of it in the future. Will get on this and respond to the other comments tomorrow! Just wanted to note my agreement on this one before bed :D

Reply:
Hah got it I was doing same thing last night :D.

Sounds good let's do it! I think it is good to try and keep AppSpec simple, so I also at first thought "hm do we really need it", but well it doesn't sound too weird / coupled that AppSpec knows a bit about where the files are coming from! So probably best to just go with it and put it in.